include/irr/asset/CTexturePacker.h

```diff
@@ -0,0 +1,107 @@
+#ifndef __IRR_C_TEXTURE_PACKER_H_INCLUDED__
+#define __IRR_C_TEXTURE_PACKER_H_INCLUDED__
+
+#include "irr/asset/format/EFormat.h"
+#include "irr/asset/ICPUImage.h"
+#include "irr/core/alloc/GeneralpurposeAddressAllocator.h"
+
+namespace irr {
+namespace asset
+{
+
+class CTexturePacker
+{
+    using addr_alctr_t = core::GeneralpurposeAddressAllocator<uint32_t>;
+
+public:
+    struct STexOffset
+    {
+        uint32_t x, y;
+    };
+
+    CTexturePacker(E_FORMAT _format, uint32_t _szxy = 1u<<14, uint32_t _pgSzxy = 256u) :
+        m_pgSzxy(_pgSzxy),
+        m_alctrReservedSpace(core::make_refctd_dynamic_array<decltype(m_alctrReservedSpace)>(
+            addr_alctr_t::reserved_size(_szxy / _pgSzxy, _szxy / _pgSzxy, 1ull))
+        ),
+        m_addrAlctr(reinterpret_cast<void*>(m_alctrReservedSpace->data()), 0u, 0u, _szxy/_pgSzxy, _szxy,_pgSzxy, 1u)
+    {
+        assert(core::isPoT(_szxy));
+        assert(core::isPoT(_pgSzxy));
+
+        ICPUImage::SCreationParams params;
+        params.extent = {_szxy,_szxy,1u};
```

Comment:
you can make it sized
```cpp
auto halfSize = pageCountBits>>1u;
{1u<<(pageCountBits-halfSize),1u<<halfSize,1u}
```

then you have slightly more increments of megatexture size